ai2thor/tests/fifo_client.py

```diff
@@ -0,0 +1,52 @@
+import ai2thor.fifo_server
+import struct
+import json
+class FifoClient:
+
+
+    def __init__(self, server_pipe_path, client_pipe_path):
+        self.server_pipe = None
+        self.client_pipe = None
+        self.server_pipe_path = server_pipe_path
+        self.client_pipe_path = client_pipe_path
+
+    def send(self, field_type, body):
+        if self.server_pipe is None:
+            self.server_pipe = open(self.server_pipe_path, "wb" )
+        
+        header = struct.pack(ai2thor.fifo_server.FifoServer.header_format, field_type, len(body))
+
+        self.server_pipe.write(header + body)
+
+    def send_eom(self):
+        header = struct.pack(ai2thor.fifo_server.FifoServer.header_format, ai2thor.fifo_server.FieldType.END_OF_MESSAGE, 0)
+        self.server_pipe.write(header)
+        self.server_pipe.flush()
+
+    def recv(self):
```

Comment:
Feels like the send and receive functionality is almost the same between the Client and Server, the client is just for testing right?

Reply:
Yeah, its just for testing. If we needed for anything else I would make it part of the main package, but I kind of like having it separate since it essentially represents an external implementation, similar to the csharp one and will help a bit to find potential bugs if we change the server but neglect to change the client.